benchmark/benchmain/main.go

```diff
@@ -350,14 +382,30 @@ func main() {
 		results = testing.BenchmarkResult{N: int(count), T: time.Now().Sub(startTime),
 			Bytes: 0, MemAllocs: memStats.Mallocs - startAllocs, MemBytes: memStats.TotalAlloc - startBytes}
 	}
+	// If features(except trace mode) are shared by all test cases, print them ahead.
+	printPos := make([]bool, len(featuresPos))
+	for i := 1; i < len(featuresPos); i++ {
+		if featuresNum[i] <= 1 {
+			printPos[i] = true
+		}
+	}
+	benchFeature := bm.Features{
+		Latency: ltc[0], Kbps: kbps[0], Mtu: mtu[0], MaxConcurrentCalls: maxConcurrentCalls[0],
+		ReqSizeBytes: reqSizeBytes[0], RespSizeBytes: respSizeBytes[0], EnableCompressor: enableCompressor[0],
+	}
+	sharedFeature := fmt.Sprintf("enable-tracing_%t\n", enableTrace[0])
+	sharedFeature += partialPrintString(printPos, benchFeature, true)
+	fmt.Println("\n Shared festures: \n" + strings.Repeat("-", 20))
+	fmt.Println(sharedFeature)
+	fmt.Println(strings.Repeat("-", 30))
+	for i := 0; i < len(featuresPos); i++ {
+		printPos[i] = !printPos[i]
+	}
+	fmt.Printf("%-80s \t%s \t%s \t%s \t%s\n", "Benchmark Name", "50% latency", "90% latency", "Bytes per Alloc", "Number of Allocs")
```

Comment:
How about a formatting function you can call that is simply:
```go
fmt.Printf("%-80s%12s%12s%12s%12s\n", args...)
```
And then:
```go
// print the header
printline("Name", "latency-50", "latency-90", "Alloc (B)", "Alloc (#)")
...
// print this benchmark's data
printline(<bm name>, <val>, <val>, <val>, <val>)
```
?

(Note that I shortened the names a bit to help them fit better)

Reply:
Done. I move this part to the benchresult/main.go because only printing table needs it. The benchmain output the raw data.